### ISSUE TITLE
[FEATURE] migration guide from spring-data-elasticsearch to spring-data-opensearch

### ISSUE THREAD

**Issue by rursprung** [opensearch-project/spring-data-opensearch#13]
### Is your feature request related to a problem?
all users were so far stuck on spring-data-elasticsearch (incompatible with OpenSearch 2.x) and will now want to switch to spring-data-opensearch (once available).

### What solution would you like?
please provide a clear step-by-step migration guide on how to migrate an existing project from spring-data-elasticsearch to spring-data-opensearch.
if it is supported (cf. also #12) also mention how a project can support both Elasticsearch and OpenSearch in parallel (only ES 7.10.2 & older; maybe even all 7.x and maybe there's even a way for a project to support OpenSearch & ES 8.x in parallel, e.g. by picking the library at runtime?)

### What alternatives have you considered?
everyone can figure it out on their own, leading to greater variation in how it's implemented and lots of questions around it.

### Do you have any additional context?
n/a

**Comment by dblock** [comment]
@rursprung It's a good idea. Any interest in contributing? I think we should have USER_GUIDE ala https://github.com/opensearch-project/opensearch-js/blob/main/USER_GUIDE.md, and either make migration a section of it, or a separate MIGRATION_GUIDE.md or UPGRADING.md.

**Comment by Neuw84** [comment]
Hi @dblock, 

I could contribute if needed :). 

Moreover, I have a project https://github.com/aws-samples/opensearch-bootful with Elastic Libraries that I will upgrade with OpenSearch libraries ( will document the process). 

Let me know!

**Comment by dblock** [comment]
@Neuw84 Amazing! Please do.

**Comment by wbeckler** [comment]
@Neuw84 The client has made big strides, now includes sigv4 and is on track for Maven soon. What's your thinking on the documentation?

**Comment by rursprung** [comment]
> When do you think that is going to be available on Maven? :)

in -2 weeks 😆: https://repo1.maven.org/maven2/org/opensearch/client/
(there's also a tag for it in git, but it hasn't been made an official GitHub release yet, so it's not that clearly visible - @reta maybe you'd want to turn that tag into a GitHub Release so it's more prominent?)

**Comment by Neuw84** [comment]
Great!

Will work on it when I have some bandwidth!

**Comment by reta** [comment]
> > When do you think that is going to be available on Maven? :)
> 
> in -2 weeks 😆: https://repo1.maven.org/maven2/org/opensearch/client/ (there's also a tag for it in git, but it hasn't been made an official GitHub release yet, so it's not that clearly visible - @reta maybe you'd want to turn that tag into a GitHub Release so it's more prominent?)

Thanks a lot for cheerful comments @rursprung ! The `0.1.0` is a pre-release, there is a one minor issue to address with test artifact name but .... things turned out to be more complicated [1], [2] so I don't know when we could have an official release :)

[1] https://github.com/opensearch-project/spring-data-opensearch/pull/75
[2] https://github.com/opensearch-project/spring-data-opensearch/pull/87

**Comment by Neuw84** [comment]
Well, will wait to that security review and the proceed. 

My idea is to rebase my repo on top on SpringBoot 3.0 and migrate all the libraries to OpenSearch ones ( and use that process for the guide).

**Comment by Neuw84** [comment]
Hi @reta, 

Seems that both issues and security review has finished. 

Will work on this issue as soon as I can 👍

**Comment by reta** [comment]
Hey @Neuw84 , thanks a lot!

**Comment by Neuw84** [comment]
@reta ,

Managed on AWS. Maybe is because it can´t communicate with the nodes directly? (Amazon OpenSearch Service just exposes an endpoint). 

Anyway, I have upgraded the code to use OpenSearch libraries and will start working on the migration guide (I also have working code with Elastic libraries). 

https://github.com/aws-samples/opensearch-bootful

Will update this post!

**Comment by reta** [comment]
@Neuw84 
> Managed on AWS. Maybe is because it can´t communicate with the nodes directly? (Amazon OpenSearch Service just exposes an endpoint).

Exactly, I believe sniffer gets internal IPs which are not accessible.

> Anyway, I have upgraded the code to use OpenSearch libraries and will start working on the migration guide (I also have working code with Elastic libraries).

Thank you very much, super helpful

**Comment by navaneeth-spotnana** [comment]
@reta I am also facing the same issue with AWS Managed Elasticsearch. Is it possible to disable the sniffer? Is that a good idea?

**Comment by N4zroth** [comment]
When migrating, am I supposed to continue using the (now deprecated) ElasticsearchRestTemplate or should I switch to OpenSearchRestTemplate?
